### PR TITLE
Tweaks and fixes

### PR DESCRIPTION
**Pull request by SuicidalInsanity**
-Fix Struct Pwing error in Flightscene
-Move Struct Pwing to structural parts tab
-Don't have missiles/bombs in a salvo fratricide each other
 -Fix DataLink VRD range
-Fix various functionalities with Armor tool
-Add penetration depth to weapons infocards
-Fix Barrage RoF for multibarrel weapons using single fireanim

### CODE REVIEW

File: BDArmory/FX/ExplosionFX.cs [hunk 1]
```diff
@@ -317,7 +317,10 @@ private List<BlastHitEvent> ProcessingBlastSphere()
                         if (partHit != null)
                         {
                             if (ProjectileUtils.IsIgnoredPart(partHit)) continue; // Ignore ignored parts.
-                            if (ExplosivePart != null && partHit.name == ExplosivePart.name) continue; //don't fratricide fellow missiles/bombs in a launched salvo when the first detonates
+                            if (ExplosivePart != null)
+                            {
+                                if (partHit.name == ExplosivePart.name) continue; //don't fratricide fellow missiles/bombs in a launched salvo when the first detonates
```
Comment (BrettRyland) [inline BDArmory/FX/ExplosionFX.cs:322]: I don't get what you changed here. The logic is the same as it was before.

File: BDArmory/Radar/VesselRadarData.cs [hunk 2]
```diff
@@ -1162,14 +1162,20 @@ private void WindowRadar(int windowID)
...
+                if (iCount > 0)
+                    DrawDisplayedIRContacts();
+                else
+                    GUI.Label(RadarDisplayRect, "NO DATA\n", lockStyle);
             }
             else
             {
                 DrawDisplayedContacts();
+                if (iCount > 0)
+                    DrawDisplayedIRContacts(); //something is throwing a GUIClips NRE somewhere before DisplayRadarControls()
```
Comment (BrettRyland) [inline BDArmory/Radar/VesselRadarData.cs:1176]: That's probably the line
```csharp
if (availableRadars.Count + availableIRSTs.Count == 0) return;
```
right after `GUI.BeginGroup(RadarDisplayRect);` due to the unbalanced `Begin`...`End`.

File: BDArmory/UI/BDAEditorArmorWindow.cs [hunk 3]
```diff
@@ -120,29 +124,30 @@ private void FillArmorList()
...
+        //private bool delayedRefreshVisuals = false;
         private bool delayedRefreshVisualsInProgress = false;
         IEnumerator DelayedRefreshVisuals(ShipConstruct ship)
         {
             delayedRefreshVisualsInProgress = true;
             var wait = new WaitForFixedUpdate();
-            while (delayedRefreshVisuals) // Wait until ship modified events stop coming.
-            {
-                delayedRefreshVisuals = false;
+            //while (delayedRefreshVisuals) // Wait until ship modified events stop coming. //preventing the remainder of the ienumerator from getting called. reverting to a simpler wai implementation  -SI
```
Comment (BrettRyland) [inline BDArmory/UI/BDAEditorArmorWindow.cs:138]: Was this never escaping the `while` loop? That should only happen if there's a continuous stream of `OnEditorShipModifiedEvent`s every fixed update, which wasn't happening before.

File: BDArmory/FX/ExplosionFX.cs (hunk 1 above)
Comment (SuicidalInsanity) [reply BDArmory/FX/ExplosionFX.cs:322]: That was I had added a Debug.Log line in there, which needed the curly braces, when I was tracking down why the SourceVessel wasn't getting updated properly, which later got removed, but without reverting back to the original configuration

Comment (BrettRyland) [reply BDArmory/FX/ExplosionFX.cs:322]: Ah ok, no worries then.

File: BDArmory/Radar/VesselRadarData.cs (hunk 2 above)
Comment (SuicidalInsanity) [reply BDArmory/Radar/VesselRadarData.cs:1176]: Ah, yeah, that would do it.

File: BDArmory/UI/BDAEditorArmorWindow.cs (hunk 3 above)
Comment (SuicidalInsanity) [reply BDArmory/UI/BDAEditorArmorWindow.cs:138]: Apparently. I was testing by adding a single part to a test ship, and removing it; old way would get caught by the delayedRefreshVisuals bool, according to debug.log lines I stuck in there for testing

File: BDArmory/Radar/VesselRadarData.cs [hunk 4]
```diff
@@ -926,7 +926,11 @@ private void WindowRadar(int windowID)
             //==============================
             GUI.BeginGroup(RadarDisplayRect);
 
-            if (availableRadars.Count + availableIRSTs.Count == 0) return;
+            if (availableRadars.Count + availableIRSTs.Count == 0)
```
Comment (BrettRyland) [inline BDArmory/Radar/VesselRadarData.cs:929]: Couldn't this just go before the `GUI.BeginGroup(RadarDisplayRect);`? It seems unnecessary to begin a group then immediately end it.

File: BDArmory/UI/BDAEditorArmorWindow.cs (hunk 3 above)
Comment (BrettRyland) [reply BDArmory/UI/BDAEditorArmorWindow.cs:138]: OK, I'll have to look into why it was doing that then. It shouldn't be getting a continuous stream of `OnEditorShipModifiedEvents`.

File: BDArmory/Radar/VesselRadarData.cs (hunk 4 above)
Comment (SuicidalInsanity) [reply BDArmory/Radar/VesselRadarData.cs]: yes

**Pull request merged**